simple_network_sim/network_of_populations/visualisation.py

```diff
@@ -0,0 +1,157 @@
+    :type df: pandas DataFrame
+    :param nodes: creates one plot per node listed (None means all nodes)
+    :type nodes: list (of region names).
+    :param states: plots one curve per state listed (None means all states)
+    :type states: list (of disease states).
+    :param ncol: number of columns (the number of rows will be calculated to fit all graphs)
+    :type ncol: int
+    :param sharey: set to true if all plots should have the same y-axis
+    :type sharey: bool
+    :param figsize: select the size of each individual plot
+    :type figsize:
+    :param cmap: color map to use
+    :type cmap:
+    :return: returns a matplotlib figure
+    :rtype: matplotlib figure
+    """
+    if nodes is None:
+        nodes = df.node.unique().tolist()
+    if states is None:
+        states = df.state.unique().tolist()
+    if cmap is None:
+        cmap = ListedColormap(["#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#999999", "#E69F00"])
+    nrow = math.ceil(len(nodes) / ncol)
+    if figsize is None:
+        figsize = (20, nrow * 5)
+
+    if not nodes:
+        raise ValueError("nodes cannot be an empty list")
+    if not states:
+        raise ValueError("states cannot be an empty list")
+
+    # pre filter by states
+    df = df[df.state.isin(states)]
+
+    fig, axes = plt.subplots(nrow, ncol, squeeze=False, constrained_layout=True, sharey=sharey, figsize=figsize)
+
+    count = 0
+    ax = None
+    for i in range(nrow):
+        for j in range(ncol):
+            if count < len(nodes):
+                node = nodes[count]
+                count += 1
+                grouped = df[df.node == node].groupby(["time", "state"]).sum()
+                indexed = grouped.reset_index().pivot(index="time", columns="state", values="total")
+
+                ax = axes[i, j]
+                indexed.plot(ax=ax, legend=False, title=node, cmap=cmap)
+                ax.set_ylabel("Number of People")
+                ax.set_xlabel("Time")
+
+    assert ax is not None, "ax was never assigned"
+    handles, labels = ax.get_legend_handles_labels()
+    fig.legend(handles, labels, loc="upper right")
+
+    return fig
+
+
+def read_output(data_product: str, path: str) -> pd.DataFrame:
+    """
+    Read a data product from the run in path
+
+    :param data_product: the name of the data_product to read
+    :param path: the path to read it from
+    :return: The output data, loaded as a pandas DataFrame
+    """
+    with open(path) as fp:
+        access_log = yaml.safe_load(fp)
+    outputs = list(
+        filter(
+            lambda x: x["type"] == "write" and x["call_metadata"]["data_product"] == data_product,
+            access_log["io"]
+        )
+    )
+    assert len(outputs) == 1, f"More than one output selected: {outputs}"
+
+    output_path = Path(access_log["data_directory"]) / Path(outputs[0]["access_metadata"]["filename"])
+    return pd.read_csv(output_path)
```

Comment:
If we factored out the config file part of the file API like we were discussing the other day then you could probably use the file/higher level API to do this directly (maybe that is one of the reasons you mentioned it).

Reply:
Yeah, I was kinda wanting to use the API to load this. But I actually quite like the idea of the data pipeline api dealing with files, and letting the models worry about the format of those files.